src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -0,0 +1,552 @@
+        public void Sealed_LangVersion()
+        {
+            const string text1 = @"
+class C
+{
+    internal virtual void M1() { }
+}
+partial class D : C
+{
+    internal sealed override partial string ToString();
+    internal sealed override partial string ToString() => ""hello"";
+}";
+            var comp = CreateCompilation(text1);
+            comp.VerifyDiagnostics(
+                // (8,45): error CS8652: The feature 'extended partial methods' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     internal sealed override partial string ToString();
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "ToString").WithArguments("extended partial methods").WithLocation(8, 45),
+                // (9,45): error CS8652: The feature 'extended partial methods' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     internal sealed override partial string ToString() => "hello";
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "ToString").WithArguments("extended partial methods").WithLocation(9, 45)
+            );
+            comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Sealed_Mismatch()
+        {
+            const string text1 = @"
+class C
+{
+    internal virtual void M1() { }
+}
+partial class D : C
+{
+    internal sealed override partial void M1();
+    internal override partial void M1() { }
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (9,36): error CS9058: Both partial method declarations must be 'sealed' or neither may be 'sealed'.
+                //     internal override partial void M1() { }
+                Diagnostic(ErrorCode.ERR_PartialMethodSealedDifference, "M1").WithLocation(9, 36)
+            );
+        }
+
+        [Fact]
+        public void Extern_LangVersion()
+        {
+            const string text1 = @"
+using System.Runtime.InteropServices;
+
+partial class C
+{
+    static partial void M1();
+
+    [DllImportAttribute(""something.dll"")]
+    static extern partial void M1();
+}";
+            var comp = CreateCompilation(text1);
+            comp.VerifyDiagnostics(
+                // (9,32): error CS8652: The feature 'extended partial methods' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     static extern partial void M1();
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M1").WithArguments("extended partial methods").WithLocation(9, 32));
+
+            comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
```

Comment:
Do we have a test that verifies the `DllImportAttribute` is emitted correctly in this scenario?

Reply:
Please also verify the IL for a call to `M1()` to ensure the call is emitted.

---
In reply to: [417430836](https://github.com/dotnet/roslyn/pull/43582#discussion_r417430836) [](ancestors = 417430836)

Reply:
Fixed